source/common/common/utility.cc

```diff
@@ -116,29 +120,35 @@ std::string DateFormatter::parse(const std::string& format_string) {
   std::string new_format_string = format_string;
   std::smatch matched;
   size_t step = 0;
-  while (regex_search(new_format_string, matched, SubsecondConstants::get().PATTERN)) {
-    const std::string& width_specifier = matched[1];
-
-    // When %f is the specifier, the width value should be 9 (the number of nanosecond digits).
-    const size_t width = width_specifier.empty() ? 9 : width_specifier.at(0) - '0';
+  while (regex_search(new_format_string, matched, SpecifierConstants::get().PATTERN)) {
+    // The std::smatch matched for (%([1-9])?f)|(%s): [all, subsecond-specifier, subsecond-specifier
+    // width, second-specifier].
+    const std::string& width_specifier = matched[2];
+    const std::string& second_specifier = matched[3];
+
+    // If %f is the specifier, the width value should be 9 (the number of nanosecond digits).
+    // While if the specifier is %s, the width of it should be 2.
```

Comment:
This seems a bit confusing, are we overloading the meaning of "width" here? In any case, it might be good to clearly state what "width" means, e.g. "In the template string to be used in runtime substitution, the width is the number of characters to be replaced."

Reply:
Updated in here b79ada9 but not sure if I have addressed your concern.